components/brave_ads/core/internal/resources/resources_util_impl.h

```diff
@@ -27,32 +27,33 @@ class File;
 namespace brave_ads::resource {
 
 template <typename T>
-std::unique_ptr<ParsingResult<T>> ReadFileAndParseResourceOnBackgroundThread(
+base::expected<T, std::string> ReadFileAndParseResourceOnBackgroundThread(
     base::File file) {
   if (!file.IsValid()) {
-    return {};
-  }
-  std::string content;
-  const base::ScopedFILE stream(base::FileToFILE(std::move(file), "rb"));
-  if (!base::ReadStreamToString(stream.get(), &content)) {
-    return {};
+    return base::unexpected("File is not valid.");
   }
 
-  absl::optional<base::Value> root = base::JSONReader::Read(content);
-  if (!root) {
-    return {};
-  }
+  absl::optional<base::Value> root;
+  {
+    std::string content;
+    const base::ScopedFILE stream(base::FileToFILE(std::move(file), "rb"));
+    if (!base::ReadStreamToString(stream.get(), &content)) {
+      return base::unexpected("Couldn't read file.");
```

Comment:
nit: we will log these values to the Brave Rewards log, for consistency it would be great if we could remove the `.`?

Reply:
Done.